Sources/Data/Microsoft.Psi.Data/Dataset.cs

```diff
@@ -145,30 +184,42 @@ public void Append(Dataset inputDataset)
                     newSession.AddStorePartition(StreamReader.Create(p.StoreName, p.StorePath, p.StreamReaderTypeName), p.Name);
                 }
             }
+
+            this.OnChangingOperation();
         }
 
         /// <summary>
         /// Saves this dataset to the specified file.
         /// </summary>
-        /// <param name="filename">The name of the file to save this dataset into.</param>
-        /// <param name="useRelativePaths">Indicates whether to use full or relative store paths.</param>
-        public void Save(string filename, bool useRelativePaths = true)
+        /// <param name="filename">The name of the file to save this dataset into (optional, default is "").</param>

```

Comment:
In general we've tried in the codebase when we have optional parameters to comment using the style "An optional blah that does blah", e.g. An optional filename that indicates the location to save the dataset, or something along these lines. 

I wonder also if for more clarity in the API we should have a Save() method that doesn't take a filename, uses this.Filename and throws if that's not set, and a SaveAs() method that takes a non-optional filename and assigns it to this.Filename and then saves.

Reply:
The split would make it clearer and hopefully less mistakes. I'll change the implementation.